tests/h/auth/util_test.py

```diff
@@ -200,6 +202,114 @@ def test_does_not_raise_when_authority_matches(self, pyramid_request, auth_clien
         util.validate_auth_client_authority(auth_client, authority)
 
 
+class TestPrincipalsForAuthClient(object):
+
+    def test_it_sets_auth_client_principal(self, auth_client):
+        principals = util.principals_for_auth_client(auth_client)
+
+        assert "auth_client:{authority}".format(authority=auth_client.authority) in principals
+
+    def test_it_sets_authority_principal(self, auth_client):
+        principals = util.principals_for_auth_client(auth_client)
+
+        assert "authority:{authority}".format(authority=auth_client.authority) in principals
+
+    def test_it_returns_principals_as_list(self, auth_client):
+        principals = util.principals_for_auth_client(auth_client)
+
+        assert isinstance(principals, list)
+
+
+class TestCheckAuthClient(object):
+    def test_it_queries_for_auth_client_in_db(self, pyramid_request):
+        pyramid_request.db.query.return_value.get.return_value = None
+        util.check_auth_client(username='whatever', password='random', request=pyramid_request)
+
+        pyramid_request.db.query.assert_called_once_with(AuthClient)
+        pyramid_request.db.query.return_value.get.assert_called_once_with('whatever')
+
+    def test_it_handles_sa_statement_exception_if_client_id_malformed(self, pyramid_request):
+        pyramid_request.db.query.return_value.get.side_effect = sa.exc.StatementError(
+            message='You did it wrong',
+            statement=None,
+            params=None,
+            orig=None)
+
+        # does not raise
+        util.check_auth_client(username='malformed', password='random', request=pyramid_request)
+
+    def test_it_returns_None_if_client_id_malformed(self, pyramid_request):
+        pyramid_request.db.query.return_value.get.side_effect = sa.exc.StatementError(
+            message='You did it wrong',
+            statement=None,
+            params=None,
+            orig=None)
+
+        # does not raise
+        principals = util.check_auth_client(username='malformed', password='random', request=pyramid_request)
+
+        assert principals is None
+
+    def test_it_returns_None_if_no_authclient_record_found_in_db(self, pyramid_request):
+        pyramid_request.db.query.return_value.get.return_value = None
+        principals = util.check_auth_client(username='whatever', password='random', request=pyramid_request)
+
+        assert principals is None
+
+    def test_it_returns_None_if_client_secret_is_None(self, pyramid_request, factories):
+        insecure_auth_client = factories.AuthClient()
+        pyramid_request.db.query.return_value.get.return_value = insecure_auth_client
+
+        principals = util.check_auth_client(username='whatever', password='random', request=pyramid_request)
+
+        assert insecure_auth_client.secret is None
```

Comment:
This first assert is meant for readability of the test more than an actual assert.